src/reducers/entities/posts.js

```diff
@@ -359,6 +359,33 @@ function handleRemovePost(posts = {}, postsInChannel = {}, postsInThread = {}, a
     return {posts: nextPosts, postsInChannel: nextPostsForChannel, postsInThread: nextPostsForThread || postsInThread};
 }
 
+function handleRethreadPost(postsMap, action) {
+    const posts = postsMap.posts ? postsMap.posts : {};
+    const postsInChannel = postsMap.postsInThread ? postsMap.postsInChannel : {};
+    const postsInThread = postsMap.postsInThread ? postsMap.postsInThread : {};
+    var post = action.data.post;
+    const nextPosts = posts;
+    nextPosts[post.id] = post;
+    const rootId = action.data.original_root_id;
+
+    const nextPostsForChannel = postsInChannel;
+    let nextPostsForThread = postsInThread;
+
+    if (nextPosts[post.id]) {
+        if (postsInThread[rootId]) {
+            nextPostsForThread = nextPostsForThread || {...postsInThread};
+            const threadPosts = [...postsInThread[rootId]];
```

Comment:
And here's where you could use the name `nextPostsForThread` since that's usually how it's used 😛